File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode(0);
        ListNode pointer = dummy;
        ListNode remove = head;

        for (int i = 0; i < n - 1; i++) {
            remove = remove.next;
        }

        while (remove.next != null) {
            pointer.next = head;
            pointer = pointer.next;
            head = head.next;
            remove = remove.next;
        }

        pointer.next = head.next;

        return dummy.next;
    }
}

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None:
            return None
        head_next = head.next
        head.next = None
        while head_next is not None:
            rest = head_next.next
            head_next.next = head
            head = head_next
            head_next = rest
        
        return head